src/assisted_test_infra/test_infra/controllers/node_controllers/oci_api_controller.py

```diff
@@ -432,10 +452,69 @@ def get_ingress_and_api_vips(self) -> dict:
 
     def destroy_all_nodes(self) -> None:
         log.info("OCI Destroying all nodes")
+        if not self._cleanup_resources:
+            self._generate_cleanup_resources()
         for resource in self._cleanup_resources[::-1]:
-            resource()
+            try:
+                resource()
+            except Exception as e:
+                log.error(f"Error during cleanup resource execution: {e}")
         pass
 
+    def _generate_cleanup_resources(self, timeout_seconds: int = 2800) -> None:
```

Comment:
@dudyas6 I'll pull your destroy logic in https://github.com/openshift/assisted-test-infra/pull/2573 as we need it for our CI running in Prow